quarkchain/cluster/shard_state.py

```diff
@@ -49,11 +49,8 @@
 
 
 class GasPriceSuggestionOracle:
-    def __init__(
-        self, last_price: int, last_head: bytes, check_blocks: int, percentile: int
-    ):
-        self.last_price = last_price
-        self.last_head = last_head
+    def __init__(self, check_blocks: int, percentile: int):
+        self.last_price = defaultdict(lambda: LRUCache(maxsize=128))
```

Comment:
why defaultdict? how about just use `LRCCache`. also consider renaming `last_price` to `cache`